content/reference/stacks.md

```diff
@@ -0,0 +1,195 @@
+
+## Terraform Workspace Names
+
+`atmos` and accompanying terraform automation modules like [terraform-spacelift-cloud-infrastructure-automation](https://github.com/cloudposse/terraform-spacelift-cloud-infrastructure-automation) will automatically create terraform [workspaces](https://www.terraform.io/docs/language/state/workspaces.html) when managing components. These workspaces derive their names from the stack name and the component name in question following this pattern: `$env-$stage-$component`. The result is workspace names like `ue2-dev-eks` or `uw2-prod-mq-broker`.
+
+<!--
+
+TODO: Do we want to define the folder structure of `stacks/` and subdirectories for catalogs here or elsewhere? I'm not fully spun up on that convention right now, so going to leave this as a todo item for discussion.
+
+## Folder Structure
+-->
+
+# Stack Schema
+
+<!--
+TODO: Is the vault backend supported as of today?
+TODO: Should we mention the `remote` backend considering the current state of the TFC automation module?
+TODO: Do we have a defined JSON schema document to share here for Stack validation?
+-->
+
+```yaml
+# stacks/ue2-dev.yaml
+
+# Imports enables shared configuration / settings across different stacks
+# The referenced files are deep merged into this stack to support granular configuration capabilities
+imports:
+  # Merge the below `stacks/*.yaml` files into this stack to provide any shared `vars` or `components.*` configuration
+  - globals
+  - ue2-globals.yaml
+  - gbl-dev.yaml
+
+# `vars` provides shared configuration for all components -- both terraform + helmfile
+vars:
+  # Used to determine the name of the workspace (e.g. the 'dev' in 'ue2-dev')
+  stage: dev
+  # Used to determine the name of the workspace (e.g. the 'ue2' in 'ue2-dev')
+  environment: ue2
+
+# Define cross-cutting terraform configuration
+terraform:
+  # `terraform.vars` provides shared configuration for terraform components
+  vars: {}
+
+  # `backend_type` + `backend` provide configuration for the terraform backend you
+  # would like to use for all components. This is typically defined in `globals.yaml`
+  backend_type: s3 # s3, remote, vault
+  backend:
+    s3:
+      encrypt: true
+      bucket: "eg-uw2-root-tfstate"
+      key: "terraform.tfstate"
+      dynamodb_table: "eg-uw2-root-tfstate-lock"
+      role_arn: "arn:aws:iam::999999999999:role/eg-gbl-root-terraform"
+      acl: "bucket-owner-full-control"
+      region: "us-east-2"
+    remote: {}
+    vault: {}
+
+# Define cross-cutting helmfile configuration
+helmfile:
+  # `helmfile.vars` provides shared configuration for terraform components
+  vars:
+    account_number: "999999999999"
+
+# Components are all the top-level units that make up this stack
+components:
+
+  # All terraform components should be listed under this section.
+  terraform:
+
+    # List one or more Terraform components here
+    first-component:
+      # Provide automation settings for this component
+      settings:
```

Comment:
Do we use `settings` for anything other than spacelift right now?

Reply:
It will also be used for `atmos` settings, but I'm not yet sure what the atmos ones are. We'll also add `vendir` settings.

Reply:
Got it. I'll leave those off for now since they are not actual, but good to know.